src/FSharp.Core/QueryExtensions.fs

```diff
@@ -17,269 +17,368 @@ open System.Linq.Expressions
-          typedefof<_ * _ * _ * _ * _ * _>,         typedefof<AnonymousObject<_, _, _, _, _, _>>
-          typedefof<_ * _ * _ * _ * _ * _ * _>,     typedefof<AnonymousObject<_, _, _, _, _, _, _>>
-          typedefof<_ * _ * _ * _ * _ * _ * _ * _>, typedefof<AnonymousObject<_, _, _, _, _, _, _, _>> |]
+    let tupleTypes =
+        [| typedefof<System.Tuple<_>>, typedefof<AnonymousObject<_>>
+           typedefof<_ * _>, typedefof<AnonymousObject<_, _>>
+           typedefof<_ * _ * _>, typedefof<AnonymousObject<_, _, _>>
+           typedefof<_ * _ * _ * _>, typedefof<AnonymousObject<_, _, _, _>>
+           typedefof<_ * _ * _ * _ * _>, typedefof<AnonymousObject<_, _, _, _, _>>
+           typedefof<_ * _ * _ * _ * _ * _>, typedefof<AnonymousObject<_, _, _, _, _, _>>
+           typedefof<_ * _ * _ * _ * _ * _ * _>, typedefof<AnonymousObject<_, _, _, _, _, _, _>>
+           typedefof<_ * _ * _ * _ * _ * _ * _ * _>, typedefof<AnonymousObject<_, _, _, _, _, _, _, _>> |]
+
     let anonObjectTypes = tupleTypes |> Array.map snd
-    let tupleToAnonTypeMap = 
-        let t = new Dictionary<Type,Type>()
-        for (k,v) in tupleTypes do t.[k] <- v
+
+    let tupleToAnonTypeMap =
+        let t = new Dictionary<Type, Type>()
+
+        for (k, v) in tupleTypes do
+            t.[k] <- v
+
         t
 
-    let anonToTupleTypeMap = 
-        let t = new Dictionary<Type,Type>()
-        for (k,v) in tupleTypes do t.[v] <- k
+    let anonToTupleTypeMap =
+        let t = new Dictionary<Type, Type>()
+
+        for (k, v) in tupleTypes do
+            t.[v] <- k
+
         t
 
 
     /// Recognize anonymous type construction written using 'new AnonymousObject(<e1>, <e2>, ...)'
-    let (|NewAnonymousObject|_|) e = 
+    let (|NewAnonymousObject|_|) e =
         match e with
-        | Patterns.NewObject(ctor,args) when 
-                 let dty = ctor.DeclaringType 
-                 dty.IsGenericType && anonToTupleTypeMap.ContainsKey (dty.GetGenericTypeDefinition()) -> 
-             Some (ctor, args)
+        | Patterns.NewObject (ctor, args) when
+            let dty = ctor.DeclaringType
+
+            dty.IsGenericType
+            && anonToTupleTypeMap.ContainsKey(dty.GetGenericTypeDefinition())
+            ->
+            Some(ctor, args)
         | _ -> None
 
-    let OneNewAnonymousObject (args:Expr list) =
+    let OneNewAnonymousObject (args: Expr list) =
         // Will fit into a single tuple type
         let typ = anonObjectTypes.[args.Length - 1]
         let typ = typ.MakeGenericType [| for a in args -> a.Type |]
         let ctor = typ.GetConstructors().[0]
-        let res = Expr.NewObject (ctor, args)
-        assert (match res with NewAnonymousObject _ -> true | _ -> false)
+        let res = Expr.NewObject(ctor, args)
+
+        assert
+            (match res with
+             | NewAnonymousObject _ -> true
+             | _ -> false)
```

Comment:
This too, looks a bit weird, I thing oneliner looked more readable).

Reply:
Matches are never in one line in Fantomas.

Reply:
@nojaf Can we keep existing one liner if they are short enough to fit on the line (or use some other metric)?

Reply:
TBH I look at cases like this as places where the original code needs improvement

Reply:
The lack of one line formatting for `match union with A -> true | B -> false` is a general concern for the style guide, as there is no other way to implement an `union.IsA` property or check, and that is very common - one of the reasons we don't expose the `IsA` properties has always been that implementing them is just a trivial match, but now there is a larger penalty for that

So I do wonder whether this shouls be a special case for fantomas, at least until the IsA properties are exposed to F#